takari-builder-common/src/main/java/io/takari/builder/internal/pathmatcher/PathNormalizer.java

```diff
@@ -16,17 +17,28 @@
  */
 public class PathNormalizer {
 
-  private static final char SEPARATOR_CHAR = '/';
+  static final char SEPARATOR_CHAR = '/';
+  private static final boolean FIXFS = File.separatorChar != SEPARATOR_CHAR;
+  static final String SEPARATOR = "/";
 
-  private final String basedir;
-  private final String basepath; // no trailing slash
+  private final String basedir; // canonical path without trailing file separatorChar
 
-  public PathNormalizer(Path basedir) {
-    this.basepath = normalize0(basedir);
-    this.basedir = this.basepath + SEPARATOR_CHAR;
+  private PathNormalizer(String basedir) {
+    this.basedir = basedir;
+  }
+
+  public static PathNormalizer create(Path basedir) {
+    return new PathNormalizer(toCanonicalPath(basedir));
+  }
+
+  public static PathNormalizer createFSRoot() {
```

Comment:
maybe call this createFSRootNormalizer?

Reply:
I think it is better to have `createNormalizer(Path)` and `createNormalizer()` overloaded methods and add javadoc that explains the difference.

Reply:
Sure. That works.

Reply:
Actually, just realized `createFSRoot()` does not do what I thought it did _and_ it behaves differently on *nix and windows. Will fix both the names and the behaviour in a separate commit.